### PR TITLE
notify item moved methods

### LINKED ISSUE #88: Implement Additional notifyDatasetChanged methods
Currently ExpandableRecyclerView implements
```java
 // Parent Changes
 notifyParentItemInserted(int parentPosition)}
 notifyParentItemRemoved(int parentPosition)}
 notifyParentItemChanged(int parentPosition)}
 notifyParentItemRangeInserted(int parentPositionStart, int itemCount)}
 // Child Changes
 notifyChildItemInserted(int parentPosition, int childPosition)}
 notifyChildItemRemoved(int parentPosition, int childPosition)}
 notifyChildItemChanged(int parentPosition, int childPosition)}
```

Would be nice to have:
```java
 notifyParentItemRangeRemoved(int parentPositionStart, int itemCount)}
 notifyParentItemRangeChanged(int parentPositionStart, int itemCount)}

 notifyChildItemRangeRemoved(int parentPosition, int childPositionStart, int itemCount)}
 notifyChildItemRangeInserted(int parentPosition, int childPositionStart, int itemCount)}
 notifyChildItemRangeChanged(int parentPosition, int childPositionStart, int itemCount)}

notifyParentItemMoved(int parentFromPosition, int parentToPosition)
notifyChildItemMoved(int parentPosition, int childFromPosition, int childToPosition)
```

### PR DESCRIPTION
**Pull request by paul-turner**
Last one of the set, so this will fix #88.

Moving the parent item is a little limited by the fact that RecyclerView.Adapter does not support a notify method that allows us to move multiple items. Instead we had to do a remove and add which gives a slightly different animation than the single move (this is only when the parent is expanded, all others used `notifyItemMoved`)

To test the child move you'll have to add child items using the "add child to end of 2nd parent" button we have.

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -44,6 +44,8 @@
     <string name="modify_two_children">Modify two children from last parent</string>
     <string name="remove_two_parents">Remove last two Parents</string>
     <string name="modify_two_parents">Modify last two Parents</string>
+    <string name="parent_move">Move parent 1 to pos 3</string>
+    <string name="child_move">Move 1st child on 2nd parent to end</string>
```
Comment (dgreenhalgh) [inline app/src/main/res/values/strings.xml]: Should use position name here - or this is moving the wrong items

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java [hunk 2]
```diff
@@ -856,6 +856,66 @@ private int changeParentWrapper(int wrapperIndex, ParentListItem parentListItem)
...
+     * <p>This is a structural change event. Representations of other existing items in the
+     * data set are still considered up to date and will not be rebound, though their
+     * positions may be altered.</p>
+     *
+     * @param fromParentPosition Previous position of the ParentListItem, relative to list of
+     *                           ParentListItems only.
+     * @param toParentPosition New position of the ParentListItem, relative to list of
+     *                         ParentListItems only.
+     */
+    public void notifyParentItemMoved(int fromParentPosition, int toParentPosition) {
```
Comment (dgreenhalgh) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: Looks like this is triggering a scroll in some cases. I'll look into it more and try to get some more information.

Comment (paul-turner) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: Yea looks like this is some weird behaviour by only the `notifyItemMoved` method of RecyclerView

**Comment by paul-turner** [comment]
Fixed the crash as pointed out by @NicolasPoirier, reworded the buttons to be a little bit more explicit wrt position

**Comment by dgreenhalgh** [comment]
Looks good :+1:

**Pull request merged**